types/sinon/sinon-tests.ts

```diff
@@ -167,8 +168,8 @@ function testXHR() {
 }
 
 function testClock() {
-    let clock = sinon.clock.create(1000);
-    clock = sinon.clock.create(new Date());
+    let clock = sinon.clock.create(1000) as fakeTimers.BrowserClock;
+    clock = sinon.clock.create(new Date()) as fakeTimers.BrowserClock;
```

Comment:
how come the cast was needed here?

Reply:
For the same reason it does in fake timers tests. They use a union of two incompatible types, so arguments of functions get an intersection to never.